src/Microsoft.AspNetCore.Cors/Infrastructure/CorsPolicyBuilder.cs

```diff
@@ -144,6 +156,17 @@ public CorsPolicyBuilder AllowAnyHeader()
         }
 
         /// <summary>
+        /// Enables the policy to be configured with wildcarded domains that will then allow
+        /// CORS from any matching subomdain
+        /// </summary>
+        /// <returns>The current policy builder</returns>
+        public CorsPolicyBuilder AllowWildcardSubdomains()
+        {
+            _policy.IsOriginAllowed = _policy.IsOriginAnAllowedSubdomain;
```

Comment:
I think combing the funcs could make this too complex. For instance should funcs be logically combined with AND or OR? 

I was in 2 minds about whether or not to provide this implementation, but figured it was better to have a robust implementation in here that was well tested to avoid others from making mistakes that might make their app insecure. However if people want to use this to different conditions they should probably take complete control of that logic to make it easier to understand. 

I will try and think of a better name for this method.